tsl/test/expected/dist_commands.out

```diff
@@ -344,3 +324,25 @@ ERROR:  data node "myserver" is not a TimescaleDB server
 \set ON_ERROR_STOP 1
 DROP SERVER myserver;
 DROP EXTENSION postgres_fdw;
+-- Test that transactional behaviour is the default and that it can be
+-- disabled.
+--
+-- In this case, we only execute it on one data node since we are
+-- creating a database and multiple creations of the database would
+-- clash when executed on the same instace.
+\c :TEST_DBNAME :ROLE_SUPERUSER
+SELECT * FROM add_data_node('dist_commands_1', host => 'localhost',
+                            database => 'dist_commands_1');
+    node_name    |   host    | port  |    database     | node_created | database_created | extension_created 
+-----------------+-----------+-------+-----------------+--------------+------------------+-------------------
+ dist_commands_1 | localhost | 55432 | dist_commands_1 | t            | t                | t
+(1 row)
+
+\set ON_ERROR_STOP 0
+CALL distributed_exec('CREATE DATABASE dist_commands_magic',
+       node_list => '{dist_commands_1}');
+ERROR:  [dist_commands_1]: CREATE DATABASE cannot run inside a transaction block
```

Comment:
This error doesn't make much sense for this case since there was no explicit transaction block. I understand why it happens, because we start a transaction block remotely, but it is still confusing.

Besides, the function _should_ be tested locally with both a transaction block and without it, using transactional=true and false.

Reply:
Added tests to test both inside and outside a transaction. Executing a non-transactional `distributed_exec` inside a transaction does not make sense, so added an error for that.